### PR TITLE
Add an inplace=False/True keyword argument to ``__call__`` on fitters to allow in-place modification of models

### PR DESCRIPTION
**Pull request by astrofrog**
This then uses this keyword argument to speed up the fitting in ``parallel_fit_dask`` since we can modify the parameters in-place without any risk.

- [x] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

File: docs/changes/modeling/17033.feature.rst
Comment (pllim) [inline docs/changes/modeling/17033.feature.rst:1]: Should this be in "perf" too?

Comment (astrofrog) [reply docs/changes/modeling/17033.feature.rst:1]: Yes I think so

Comment (astrofrog) [reply docs/changes/modeling/17033.feature.rst:1]: Done

**Comment by Cadair** [comment]
pre-commit.ci autofix

**Review by mhvk — COMMENTED** [review]
This seems like the nicer approach! I guess one small suggestion (but might as well wait for proper review), is to add a `*` to ensure the new argument is keyword-only.

**Comment by larrybradley** [comment]
I'm not a `modeling` maintainer, but I like this addition!

**Comment by astrofrog** [comment]
> This seems like the nicer approach! I guess one small suggestion (but might as well wait for proper review), is to add a `*` to ensure the new argument is keyword-only.

Done!

**Comment by nden** [comment]
This looks good. A couple of comments
- Can you edit the `Returns` sections in the docstrings to clarify that the returned model is not always a copy
- Should the `inplace` parameter be passed to the `FittingWithOutlierRemoval`? Speeding up this one will be a good thing too.

**Comment by nden** [comment]
Also I wonder if a note should be added to the `Performance tips` section in the documentation.
https://docs.astropy.org/en/latest/modeling/performance.html

**Comment by astrofrog** [comment]
@nden - I've now:

* Added a performance tip
* Updated the Returns sections in the docstrings for all the fitters
* Added an explicit inplace= option to ``FittingWithOutlierRemoval`` (in fact the option was passed through anyway via ``**kwargs`` but I think it's clearer this way

**Comment by astrofrog** [comment]
pre-commit.ci autofix

**Comment by astrofrog** [comment]
@nden - this should be ready, the CI failures are unrelated.

**Comment by astrofrog** [comment]
Rebased

**Review by nden — APPROVED** [review]

**Pull request merged**
